weedcoco/importers/mask.py

```diff
@@ -0,0 +1,117 @@
+from collections import defaultdict
+from pathlib import Path
+from typing import Optional, Mapping
+import argparse
+import json
+
+from lxml import etree
+
+import cv2 as cv
+import numpy as np
+import os
+
+from weedcoco.validation import validate
+from weedcoco.utils import load_json_or_yaml
+from weedcoco.utils import add_agcontext_from_file
+from weedcoco.utils import add_collection_from_file
+
+# TODO: extend functionality to read categories from a file
+categories = [
+    {"id": 0, "name": "crop: zingiber officnale"},
+]
+
+
+def generate_masks_contours(mask_path):
+
+    """
+    Return contours' matrix of mask or image using opencv
+    """
+
+    def np_tolist(np_array):
+        return np_array.tolist()
+
+    im = cv.imread(mask_path)
+    imgray = cv.cvtColor(im, cv.COLOR_BGR2GRAY)
+    ret, thresh = cv.threshold(imgray, 127, 255, 0)
+    contours, hierarchy = cv.findContours(thresh, cv.RETR_TREE, cv.CHAIN_APPROX_SIMPLE)
+    return json.dumps(list(map(np_tolist, contours)))
+
+
+def mask_to_coco(image_dir: Path, mask_dir: Path):
+
+    """Converts images and masks to MS COCO images and annotations
+
+    Parameters
+    ----------
+    image_dir: pathlib.Path
+    mask_dir: pathlib.Path
+
+    Returns
+    -------
+    dict
+        Keys present should be 'images', 'annotations', and
+        'categories'.
+        Other COCO components should be added as a post-process.
+    """
+
+    images = []
+    annotations = []
+
+    image_id = 0
+    for filename in os.listdir(image_dir):
+        if filename.endswith(".png") or filename.endswith(".jpg"):
```

Comment:
there should be an else clause that warns or raises an error if the file type is unexpected

Reply:
`jpeg` and `tif` might also be possible extensions

Reply:
A neat shorthand
```suggestion
        if filename.endswith((".png", ".jpg", ".jpeg", ".tif", ".tiff")):
```